src/pipeline/mod.rs

```diff
@@ -9,21 +10,24 @@ pub mod ingest;
 pub mod monitor;
 
 pub async fn run(config: Config, tx_storage: Arc<SqliteTransaction>) -> Result<()> {
-    tokio::spawn(async move {
-        let ingest = ingest::Stage::new(tx_storage.clone());
-        let fanout = fanout::Stage::new(tx_storage.clone(), config.peer_manager);
-        let monitor = monitor::Stage {};
+    let ingest = ingest::Stage::new(tx_storage.clone());
+    let fanout = fanout::Stage::new(tx_storage.clone(), config.peer_manager);
 
-        let policy: gasket::runtime::Policy = Default::default();
+    let adapter = UtxoChainSyncAdapter::new(config.monitor_utxorpc);
 
-        let ingest = gasket::runtime::spawn_stage(ingest, policy.clone());
-        let fanout = gasket::runtime::spawn_stage(fanout, policy.clone());
-        let monitor = gasket::runtime::spawn_stage(monitor, policy.clone());
+    #[cfg(feature = "file")]
+    let adapter = monitor::file::FileChainSyncAdapter::try_new()?;
```

Comment:
This file adapter is meant to be used in test. We don't need to have it as part of the pipeline setup. It will be only used in testing code.